pkg/api/spark/v1/api.go

```diff
@@ -0,0 +1,190 @@
+package spark_v1
+
+import (
+	"context"
+)
+
+//go:generate mockgen -destination=./test/mock_context.go -package spark_v1_mock github.com/azarc-io/vth-faas-sdk-go/pkg/api/spark/v1 Context
+//go:generate mockgen -destination=./test/mock_stageprogress.go -package=spark_v1_mock github.com/azarc-io/vth-faas-sdk-go/pkg/api/spark/v1 StageProgressHandler
+
+/************************************************************************/
+// BUILDER
+/************************************************************************/
+
+type (
+	// Builder contract for the chain builder
+	Builder interface {
+		NewChain(name string) BuilderChain
+		ChainFinalizer
+	}
+
+	BuilderChain interface {
+		ChainNode
```

Comment:
The `BuilderChain` should enforce that it has at least one stage before allowing the other chaining funcs such as `Compensate` etc.

```suggestion
		Stage(name string, stageDefinitionFn StageDefinitionFn, options ...StageOption) ChainStage
```

Reply:
done